contracts/src/chainlink/cairo/ocr2/aggregator.cairo

```diff
@@ -1133,10 +1137,18 @@ end
 
 # --- Transmitter Payment
 
-func calculate_reimbursement() -> (amount : felt):
-    # TODO:
-    let amount = 0
-    return (amount)
+const MARGIN = 115
+
+func calculate_reimbursement{range_check_ptr}(
```

Comment:
Hmm, some thinking...

1. Static gas estimates in contracts are brittle. Gas values could change unexpectedly with a network upgrade and our estimates are off at that point (would require feed redeploys).
2. This is not an issue with on-chain gas metering, but that is only available on EVM.
3. I feel we can and probably should do the estimates off-chain. The only thing I worry it would slow down the libocr's runtime report generation hot path.
4. Instead of reporting on `gas_price` we should report the `max_fee` as this is the API the StarkNet network uses. It would represent the minimum `max_fee` as estimated by the cluster, which the transmitter should use (it's agreed upon). Transmitter is disallowed to use a fee less than agreed (implicitly for now, alerting system can flag) but it is allowed to use a higher fee if conditions changed while the transaction is in/pre flight.


So my ad-hoc suggestion would be something like this:

1. We move estimates off-chain as that way they can be cheaper (no gas spent to estimate gas) and safer (not static but dynamic on any gas costs upgrades + simpler on-chain code).
2. The DON reports on "minimum fee/cost" (signed) and the transmitter reports on his "used fee/cost" (unverified). We add the `max_fee_proposed` and `max_fee_used` to the `transmit` input, `max_fee_proposed` is used as part of report, `max_fee_used` is just logged as part of the 'new_transmission' event.
3. We cap the `max_fee_used` to be at most something like 2x over `max_fee_proposed`, so if someone decides to drain they will need to do it slowly until we detect it and kick them out.
4. We reimburse the transmitter using the `max_fee_used` (optimistically trust)
5. We add the alert condition to the on-chain monitor service, which would alert on "transmitter spent less than 'max_fee_proposed'" and "transmitter reported invalid 'max_fee_used'".

cc @aalu1418 any thoughts? This is also important in the context of TXM, because for example if a cluster agrees on a static single price/fee for a transmitter to use this would mean no gas bumping without re-agreeing and reaching consensus on a new price to use. Could make the protocol a lot slower at times.